### ISSUE TITLE
DimensionMismatch: All data containers must have the same number of observations.

### ISSUE THREAD

**Issue by CaiYitao** [FluxML/Flux.jl#1941]
I have got a dimension mismatch error as following:
```julia
args = Args()
trainloader = DataLoader((X_train,y_train); batchsize = args.batch_size, shuffle = true)

```
```julia
DimensionMismatch: All data containers must have the same number of observations.

_check_numobs_error()@observation.jl:115
_check_numobs@observation.jl:124[inlined]
numobs@observation.jl:129[inlined]
var"#DataLoader#63"(::Int64, ::Bool, ::Bool, ::Random._GLOBAL_RNG, ::Type{MLUtils.DataLoader}, ::Tuple{Matrix{Float32}, Matrix{Float32}})@dataloader.jl:77
top-level scope@[Local: 3](http://localhost:1234/edit?id=ea89c79e-b9be-11ec-071f-01779ea7482d#)[inlined]
```

I noticed that there is an issue have an answer with : Flux expects the last dimension to be the observations dim.

but I did not get it .

my X_train is a 12053×1024 Matrix{Float32}, y_train is a 12053×12 Matrix{Float32}
do you mean they need to be transposed?  like  X_train to be 1024×12053 adjoint(::Matrix{Float32})  y_train to be 12×12053 adjoint(::Matrix{Float32})

Could you please elaborate on this?

**Comment by ToucheSir** [comment]
Flip the order of the 12053 with the other number and you'll have the right dimensions. Flux expects that to be the last one, whereas right now you have it as the first one (which Python frameworks expect, but this isn't Python :) )

**Comment by CaiYitao** [comment]
Ah, thank you very much for your lighting speed response : )

**Comment by mcabbott** [comment]
Might be nice if that error message explained a bit more? Like "same number of observations. By default these are the last dimensions of arrays, got size(A) = .... and size(B) = ..."

**Comment by CaiYitao** [comment]
yeah, that would be more helpful than with only the same number of observations

**Comment by CarloLucibello** [comment]
@CaiYitao you are welcome to fila a PR to MLUtils.jl if you want to
https://github.com/JuliaML/MLUtils.jl/blob/1da3c53c1b6a5c2e4ce51ab74df358f02c17d1bf/src/observation.jl#L116

**Comment by CaiYitao** [comment]
Thank you, Let me get familiar with how this works first, when I have enough experience, I am happy to file a PR.(I haven't done that before : ) as a freshman in programming)
